src/devices_models/devices/common/timeseries_constraint.jl

```diff
@@ -225,52 +234,53 @@ Constructs lower bound for given variable using a parameter. The constraint is
 
 # Constraint
 
-``` ts_data[name].multiplier * param[name, t] <= variable[name, t] ```
+``` range_constraints[name].multiplier * param[name, t] <= variable[name, t] ```
 
 # LaTeX
 
 `` r^{val} x^{param}_t \leq x^{var}_t, \forall t ``
 
 # Arguments
 * psi_container::PSIContainer : the psi_container model built in PowerSimulations
-* ts_data::Vector{DeviceTimeSeries} : container of device time series data and scaling factors
+* range_constraints::Vector{DeviceTimeSeriesRangeConstraint} : container of device time series data and scaling factors
 * cons_name::Symbol : name of the constraint
 * param_reference::UpdateRef : UpdateRef to access the parameter
 * var_name::Symbol : the name of the variable
 """
 function device_timeseries_param_lb(
     psi_container::PSIContainer,
-    ts_data::Vector{DeviceTimeSeries},
+    range_constraints::Vector{DeviceTimeSeriesRangeConstraint},
     cons_name::Symbol,
     param_reference::UpdateRef,
     var_name::Symbol,
 )
     time_steps = model_time_steps(psi_container)
     variable = get_variable(psi_container, var_name)
     lb_name = middle_rename(cons_name, PSI_NAME_DELIMITER, "lb")
-    names = (d.name for d in ts_data)
+    names = [get_name(x) for x in range_constraints]
```

Comment:
why revert to an array?

Reply:
I thought I had a good reason.  Reverted.